### PR TITLE
Refactor MealPlanDAO into AbstractMealPlanDAO to allow for Test DAO

### PR DESCRIPTION
**Pull request by gmtcrary**
CO-831

This PR adds useTestDAO as a boolean flag in the configuration.yaml to let the developer to use either the TestMealPlanDAO or the contrib MealPlanDAO.

### CODE REVIEW

**Comment by jaredkosanovic** [comment]
Could you update the gitignore to ignore src/main/groovy/edu/oregonstate/mist/contrib/

**Comment by gmtcrary** [comment]
The contrib ignore line has been added.

File: src/main/groovy/edu/oregonstate/mist/mealcard/resources/MealPlanResource.groovy [hunk 1]
```diff
@@ -25,13 +25,13 @@ import javax.ws.rs.core.Response
...
 
-    private MealPlanDAO mealPlanDAO
+    private AbstractMealPlanDAO mealPlanDAO
     private URI endpointUri
 
     @Context
     UriInfo uriInfo
 
-    MealPlanResource(MealPlanDAO dao, URI endpointUri) {
+    MealPlanResource(AbstractMealPlanDAO dao , URI endpointUri) {
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/mealcard/resources/MealPlanResource.groovy]: Unneeded whitespace after "dao"?

Comment (gmtcrary) [reply src/main/groovy/edu/oregonstate/mist/mealcard/resources/MealPlanResource.groovy]: Fixed

**Pull request merged**